### ISSUE TITLE
E95: Buffer with this name already exists when Man is invoked as a $MANPAGER

### ISSUE THREAD

**Issue by Mo-tmd** [neovim/neovim#30132]
### Problem

I invoke Man as a $MANPAGER from a terminal within neovim.
I do this by using [neovim-remote](https://github.com/mhinz/neovim-remote) plugin and setting `MANPAGER="nvr +'Man!' -"`
`init_pager` in `man.lua` crashes if there's already an open buffer for the man pages.

The line responsible for the error is this one:
`vim.cmd.file({ 'man://' .. fn.fnameescape(ref):lower(), mods = { silent = true } })`
It tries to rename the buffer to a name that's already taken.

Suggested solution:
Check if there's already a buffer with that name and go to that existing buffer if so. 

### Steps to reproduce

* Add [neovim-remote](https://github.com/mhinz/neovim-remote) plugin.
* Add this line to your **init.vim**: `let $MANPAGER="nvr +'Man!' -"`
* Start a terminal: `:term`
* Run `man ls`
* Go back to the terminal
* Run `man ls` again

I don't know if it's possible to use neovim's builtin remote functionality for $MANPAGER. If it is possible, then the error could be reproduced without any plugins.

### Expected behavior

Don't crash and go to the existing buffer instead

### Neovim version (nvim -v)

0.9.5

### Vim (not Nvim) behaves the same?

No Man in vim

### Operating system/version

N/A

### Terminal name/version

N/A

### $TERM environment variable

N/A

### Installation

N/A

**Comment by clason** [comment]
You should just use `MANPAGER 'nvim'  '+Man!'`. Also, please use the latest version (0.10.1), at least for testing.

**Comment by Mo-tmd** [comment]
> You should just use nvim --server $NVIM --remote

I tried `let $MANPAGER="nvim --server $NVIM --remote"`. It starts neovim from the terminal within neovim

> Also, please use the latest version (0.10.1), at least for testing

Pretty sure version is irrelevant as the code responsible hasn't changed

**Comment by glacambre** [comment]
Are you sure you're seeing a crash? I think the following steps should reproduce what you see, but there's no crash, only an error message:
```
:new
:r!man -Pcat printf
:Man!
:new
:r!man -Pcat printf
:Man!
```
This prints:
```
Error executing Lua callback: /tmp/.mount_nvimMpq0c5/usr/share/nvim/runtime/lua/man.lua:675: Vim:E95: Buffer with this name already exists
stack traceback:
        [C]: in function 'file'
        /tmp/.mount_nvimMpq0c5/usr/share/nvim/runtime/lua/man.lua:675: in function 'init_pager'
        .../.mount_nvimMpq0c5/usr/share/nvim/runtime/plugin/man.lua:9: in function <.../.mount_nvimMpq0c5/usr/share/nvim/runtime/plugin/man.lua:6>
```
Which is pretty reasonable IMO.

**Comment by Mo-tmd** [comment]
@glacambre, yes that's the error I see, thank you and sorry for my confusing terminology.

Not sure it's reasonable to get an error when there's a forgotten buffer for that MAN page. It should either go to existing buffer (and wipe out the temporary one), or it should make unique buffer names (e.g. by prefixing with buffer number)

Also when using Man as a general $PAGER or $GIT_PAGER, you can get the same buffer name but with different content depending on the command. But perhaps Man plugin wasn't designed for such use cases though.

**Comment by glacambre** [comment]
>It should either go to existing buffer

I'm not sure this is possible, since two buffers for `printf` could have different contents.

>it should make unique buffer names

That sounds like a reasonable enhancement. Hopefully there aren't any plugins relying either on the current naming scheme of `man://` buffers or on catching the errors.

**Comment by justinmk** [comment]
> That sounds like a reasonable enhancement. Hopefully there aren't any plugins relying either on the current naming scheme of `man://` buffers or on catching the errors.

We don't need to change the naming scheme. We can special-case this scenario and choose a non-existing buf name to support this edge case.

Changing the `man://` URI format is out of scope.
